src/earn/saga.ts

```diff
@@ -456,7 +463,132 @@ export function* withdrawSubmitSaga(action: PayloadAction<WithdrawInfo>) {
   }
 }
 
+export function* claimSubmitSaga(action: PayloadAction<WithdrawInfo>) {
```

Comment:
I disagree with the choice to make this its own function. The vast majority of it is copy/paste and doing the exact steps as withdrawSubmitSaga. I don't think there is a strong motivation to separate them for separation of concerns, because their concerns are very close. Code-wise it looks like you will need to add two if statements max to withdrawSubmitSaga to be able to make it optionally include withdraw transactions or not.

Reply:
Refactored in 7f56a57b7d7acac488eeb66703612e4f02f504eb tests added in 1cb1d86b8b65d63fb847351213a6eae186652538.